odoo/fields.py

```diff
@@ -1783,10 +1785,25 @@ def read(self, records):
         for record in records:
             cache.set(record, self, data.get(record.id, False))
 
+    def _check_svg(self, record, value):
+        """Detect if the binary content is an SVG for restricting its upload
+        only to system users.
+        """
+        if not value:
+            return
+        if value[:1] == b'P':  # Fast detection of first 6 bits of '<' (0x3C)
+            decoded_value = base64.b64decode(value)
+            # Full mimetype detection
+            if (guess_mimetype(decoded_value) == 'image/svg+xml' and
+                    not record.user_has_groups('base.group_system')):
```

Comment:
Could use ` env.user._is_admin()` as in the `ir.attachment` neutering [here](https://github.com/odoo/odoo/blob/f04ede4ec886d5e8dc60e2d3214dcaa86ced54c8/odoo/addons/base/models/ir_attachment.py#L254-L261)